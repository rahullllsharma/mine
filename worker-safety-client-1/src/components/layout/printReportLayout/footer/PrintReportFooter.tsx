type PrintReportFooterProps = {
  note?: string;
};

/**
 * Footer component for the Daily Report PDF.
 *
 * Since this is passed as a html string, it doesn't inherit any styling properties from the inner document like styles nor custom fonts.
 * Thus we need to set everything (font-size, font-family, etc) and avoid having relative units like rem (em is ok tho).
 */
export default function PrintReportFooter({
  note,
}: PrintReportFooterProps): JSX.Element {
  return (
    <footer className="flex items-center text-neutral-shade-100 font-sans font-normal text-base px-[12px] w-full">
      {note && <p className="text-xs">{note}</p>}
      {/*
        pageNumber is a special class from puppeteer to display the page number.
        https://pptr.dev/api/puppeteer.pdfoptions.headertemplate/
      */}
      <p
        data-testid="print-report-footer-pagination"
        className="text-tiny ml-auto pageNumber"
      ></p>
    </footer>
  );
}
