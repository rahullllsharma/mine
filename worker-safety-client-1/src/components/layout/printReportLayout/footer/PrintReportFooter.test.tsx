import { render, screen } from "@testing-library/react";
import PrintReportFooter from "./PrintReportFooter";

describe(PrintReportFooter.name, () => {
  describe('when it renders without "note" and "pageCount"', () => {
    it("should match the snapshot", () => {
      const { asFragment } = render(<PrintReportFooter />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it("should include an element with the pageNumber class (used in puppeteer)", () => {
    render(<PrintReportFooter />);
    expect(screen.getByTestId("print-report-footer-pagination")).toHaveClass(
      "pageNumber"
    );
  });

  describe('when it renders with "note"', () => {
    it("should display the note element", () => {
      render(<PrintReportFooter note="This is a note" />);
      screen.getByText("This is a note");
    });
  });
});
