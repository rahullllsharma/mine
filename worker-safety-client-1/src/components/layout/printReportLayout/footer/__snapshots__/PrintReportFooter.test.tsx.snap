// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrintReportFooter when it renders without "note" and "pageCount" should match the snapshot 1`] = `
<DocumentFragment>
  <footer
    class="flex items-center text-neutral-shade-100 font-sans font-normal text-base px-[12px] w-full"
  >
    <p
      class="text-tiny ml-auto pageNumber"
      data-testid="print-report-footer-pagination"
    />
  </footer>
</DocumentFragment>
`;
